app/components/toDoListController.js

```diff
@@ -0,0 +1,30 @@
+toDoTaskList.controller('ToDoListController', [function() {
+
+  var self = this;
+  self.toDoList = [];
+
+  self.addToDo = function() {
+    if(!self.newToDo || self.newToDo === '') {
+      return;
+    }
+    self.toDoList.push( {'task': self.newToDo, 'completed': false} );
+    self.newToDo = '';
+  };
+
+  self.taskCounter = function () {
+    return self.toDoList.length;
+  };
+
+  self.deleteTask = function(item) {
+    var index = self.toDoList.indexOf(item);
+    self.toDoList.splice(index, 1);
+  };
+
+// convert to a function using filter without underscore js library
+  self.removeCompleteTasks = function() {
+    self.toDoList = _.filter(self.toDoList, function(todo){
```

Comment:
@ptolemybarnes, I did indeed, late on Sunday night, I felt pretty bad about it and if it helps it is in my readme to refactor/todo. Thanks for the link. I enjoyed this challenge so will be revisiting it and reworking.